airlearner/airlearner-utils/src/main/scala/com/airbnb/common/ml/util/HiveUtil.scala

```diff
@@ -4,15 +4,54 @@ import scala.reflect.ClassTag
 import scala.util.Try
 
 import com.typesafe.config.Config
-import org.apache.hadoop.conf.Configuration
-import org.apache.hadoop.fs.Path
 import org.apache.spark.rdd.RDD
 import org.apache.spark.sql.hive.HiveContext
 import org.apache.spark.sql.{DataFrame, Row}
 
-
 object HiveUtil {
 
+  /*
+   * Drop the specified Hive partition if it exists
+   */
+  def dropHivePartition(
+      hc: HiveContext,
+      hiveTableName: String,
+      hivePartitionSpecs: Map[String, Any]
+  ): Unit = {
+    if (!hiveTableName.contains('.')) {
+      throw new RuntimeException(s"Missing namespace for hive table: $hiveTableName.")
+    }
+
+    // Break the table name into namespace.table_name
+    val Array(namespace, table) = hiveTableName.split('.')
+    // Turn the partition spec map into a Hive-format String
+    val partitionSpec = hivePartitionSpecsMapToString(hivePartitionSpecs)
+
+    // Drop the partition if it exists
+    hc.sql(s"USE $namespace")
+    hc.sql(s"ALTER TABLE $table DROP IF EXISTS PARTITION ($partitionSpec)")
```

Comment:
This function is just `dropHivePartition(hc, hiveTableName, hivePartitionSpecsMapToString(hivePartitionSpecs))`?